Firestore/core/test/firebase/firestore/util/grpc_tests_util.h

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_TEST_FIREBASE_FIRESTORE_UTIL_GRPC_TESTS_UTIL_H_
+#define FIRESTORE_CORE_TEST_FIREBASE_FIRESTORE_UTIL_GRPC_TESTS_UTIL_H_
+
+#include <initializer_list>
+#include <memory>
+#include <utility>
+
+#include "Firestore/core/src/firebase/firestore/remote/grpc_completion.h"
+#include "Firestore/core/src/firebase/firestore/remote/grpc_stream.h"
+#include "Firestore/core/src/firebase/firestore/util/async_queue.h"
+#include "Firestore/core/src/firebase/firestore/util/executor_std.h"
+#include "absl/memory/memory.h"
+#include "grpcpp/client_context.h"
+#include "grpcpp/completion_queue.h"
+#include "grpcpp/create_channel.h"
+#include "grpcpp/generic/generic_stub.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+enum OperationResult { Ok, Error };
+
+/**
+ * Does the somewhat complicated setup required to create a `GrpcStream` and
+ * allows imitating the normal completion of `GrpcStreamOperation`s.
+ */
+class GrpcStreamFixture {
```

Comment:
In general it's useful to name classes and the files containing them after each other. Consider putting this in `grpc_stream_fixture.h`.

However, since this doesn't extend testing::Test it's not really a gtest fixture per se. You might consider renaming this to more directly name what it does such that it makes sense for an actual test fixture to have a has-a relationship with this class. `GrpcStreamTester`, maybe? (And then rename the file to match that.)

Reply:
Sorry, I forgot that gTest terminology also uses "fixtures". Changed to `GrpcStreamTester`.